Library/Homebrew/formula_installer.rb

```diff
@@ -110,6 +110,16 @@ def pour_bottle?(install_bottle_options = { warn: false })
       return false
     end
 
+    if formula.deps.any? { |dep| dep.closest_spec_for_dependency_upgrade_sym != :stable }
+      if install_bottle_options[:warn]
+        opoo <<-EOS.undent
+          Building #{formula.full_name} from source:
+            The bottle needs stable dependencies which are not installed.
```

Comment:
Should we say which ones? (e.g. in case the user wants to control+C and uninstall the offending formula)